api/capi/src/nnstreamer-capi-single.c

```diff
@@ -738,8 +738,18 @@ ml_single_invoke (ml_single_h single,
 
   check_feature_state ();
 
-  if (!single || !input || !output) {
-    ml_loge ("The given param is invalid.");
+  if (!single) {
+    ml_loge ("The first param, single is not valid.");
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+
+  if (!input) {
+    ml_loge ("The second param, input is not valid.");
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+
+  if (!output) {
+    ml_loge ("The third param, output is not valid.");
```

Comment:
IMO, messages indicating the first, second, and third parameters are invalid are hardly helpful for the application developers. Although the previous message is also about the wrong parmaters, you need to improve it.